### ISSUE TITLE
How to replace deprecated "new Context(Device.TYPE_GPU)"

### ISSUE THREAD

**Issue by Daniel-Alievsky** [bytedeco/javacpp#645]
I used the constructor "new Context(int dtype)" to get information about installed GPU devices. However, in current version this constructor is deprecated. How can I replace this call in modern API?

**Comment by Daniel-Alievsky** [comment]
Yes, OpenCV. Maybe you know, how can I replace my simplest call Context context = new Context(Device.TYPE_GPU) ?

For Context.create(String), which string should be here?

**Comment by saudet** [comment]
My guess is that we can loop over the devices returned by `Context.getDefault()`.

**Comment by Daniel-Alievsky** [comment]
Right, but I don't see a simple way to detect which devices support GPU. TYPE_GPU was a convenient flag. Is it deprecated at all?

**Comment by saudet** [comment]
That's just a flag. We can easily use the API from OpenCL to query devices:
https://github.com/opencv/opencv/blob/4.x/samples/android/tutorial-4-opencl/jni/CLprocessor.cpp#L113
The C++ API of OpenCL hasn't been mapped (yet), but the C API is available:
http://bytedeco.org/javacpp-presets/opencl/apidocs/
